### PR TITLE
feat(mobile-prompt) request project endpoint

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a new endpoint to request a someone else set up a project:
```
POST /organizations/<organization_slug>/request-project-creation/
```

This endpoint takes just one parameter: `targetUserEmail`. 

This endpoint is going to be used when we detect an organization that has requests coming from a mobile app but doesn't have a mobile project set up. We will show a CTA to the user asking them to set up a project or ask someone else to set it up. Note that part of the goal is to get new users which means we might send this to non-Sentry users as well which is why the endpoint only takes an email address instead of a user id. At some point, we may do a more sophisticated flow that combines requesting/inviting a member to join AND requesting a new project. Also, this endpoint may be modified to work with other types of project creation flows in the future. We would add a new parameter to denote the type of project to be created but for now, the only use case is mobile so need to add the parameter yet.

Screen Shot 2021-02-11 at 4 44 53 PM

### CODE REVIEW

File: src/sentry/api/endpoints/organization_request_project_creation.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.utils.email import MessageBuilder
+from sentry.utils.http import absolute_uri
+
+
+class OrganizationRequestProjectCreationSerializer(CamelSnakeSerializer):
+    target_user_email = serializers.EmailField(required=True)
+
+
+class OrganizationRequestProjectCreationPermission(OrganizationPermission):
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_request_project_creation.py:15]: Hmm, do we have a permission that already matches this?

Comment (EvanPurkhiser) [reply src/sentry/api/endpoints/organization_request_project_creation.py:15]: Like do we have to define this class, or is there something in api.bases we can use. Looks like maybe not

Comment (scefali) [reply src/sentry/api/endpoints/organization_request_project_creation.py:15]: @EvanPurkhiser Yea, I'm sure we do some places but generally `POST` endpoints require some sort of write permission.

File: src/sentry/api/endpoints/organization_request_project_creation.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        Send an email requesting a project be created
+        """
+
+        serializer = OrganizationRequestProjectCreationSerializer(data=request.data)
+        if not serializer.is_valid():
+            return self.respond(serializer.errors, status=400)
+
+        requester_name = request.user.get_display_name()
+        requester_link = absolute_uri(
+            f"/organizations/{organization.slug}/projects/new/?referrer=request_project&category=mobile"
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_request_project_creation.py:36]: Can you use `reverse` for this?

File: src/sentry/api/endpoints/organization_request_project_creation.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        )
+
+        msg = MessageBuilder(
+            subject=f"{requester_name} thinks Sentry can help monitor your mobile app",
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_request_project_creation.py:40]: Can we use `_()` translation util here?

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**